packages/lesswrong/lib/collections/revisions/collection.ts

```diff
@@ -25,7 +25,9 @@ Revisions.checkAccess = async (user: DbUser|null, revision: DbRevision, context:
   if (!revision) return false
   if ((user && user._id) === revision.userId) return true
   if (userCanDo(user, 'posts.view.all')) return true
-  
+  // not sure why some revisions have no collectionName,
+  // but this will cause an error below so just exclude them
+  if (!revision.collectionName) return false
```

Comment:
I tested that this fixes the email bug, and I tested that posts are still viewable for the author/logged in users/logged out users. Let me know if you think I should test anything else.